src/sentry/integrations/msteams/utils.py

```diff
@@ -54,3 +55,376 @@ def get_channel_id(organization, integration_id, name):
+        {"elementId": "assignSubmit"},
+    ]
+
+    if status == GroupStatus.RESOLVED:
+        resolve_action = {
+            "type": "Action.Submit",
+            "title": "Unresolve",
+            "data": {"actionType": "unresolve"},
+        }
+    else:
+        resolve_action = {
+            "type": "Action.ToggleVisibility",
+            "title": "Resolve",
+            "targetElements": resolve_targets,
+        }
+
+    if status == GroupStatus.IGNORED:
+        ignore_action = {
+            "type": "Action.Submit",
+            "title": "Stop Ignoring",
+            "data": {"actionType": "unignore"},
+        }
+    else:
+        ignore_action = {
+            "type": "Action.ToggleVisibility",
+            "title": "Ignore",
+            "targetElements": ignore_targets,
+        }
+
+    assign_text = "Assign"
+    assign_action = {
+        "type": "Action.ToggleVisibility",
+        "title": assign_text,
+        "targetElements": assign_targets,
+    }
+
+    return {
+        "type": "ColumnSet",
+        "columns": [
+            {
+                "type": "Column",
+                "items": [{"type": "ActionSet", "actions": [resolve_action]}],
+                "width": "stretch",
+            },
+            {
+                "type": "Column",
+                "items": [{"type": "ActionSet", "actions": [ignore_action]}],
+                "width": "stretch",
+            },
+            {
+                "type": "Column",
+                "items": [{"type": "ActionSet", "actions": [assign_action]}],
+                "width": "stretch",
+            },
+        ],
+    }
+
+
+def build_incident_resolve_card():
+    title_card = {
+        "type": "TextBlock",
+        "size": "Large",
+        "text": "Resolve",
+        "weight": "Bolder",
+        "id": "resolveTitle",
+        "isVisible": False,
+    }
+
+    input_card = {
+        "type": "Input.ChoiceSet",
+        "value": "0",
+        "id": "resolveInput",
+        "isVisible": False,
+        "choices": [
+            {"title": "Immediately", "value": "0"},
+            {"title": "In the current release", "value": "1"},
+            {"title": "In the next release", "value": "2"},
+        ],
+    }
+
+    submit_card = {
+        "type": "ActionSet",
+        "id": "resolveSubmit",
+        "isVisible": False,
+        "actions": [
+            {"type": "Action.Submit", "title": "Resolve", "data": {"actionType": "resolve"}}
+        ],
+    }
+
+    return [title_card, input_card, submit_card]
+
+
+def build_incident_ignore_card():
+    title_card = {
+        "type": "TextBlock",
+        "size": "Large",
+        "text": "Ignore until this happens again...",
+        "weight": "Bolder",
+        "id": "ignoreTitle",
+        "isVisible": False,
+    }
+
+    input_card = {
+        "type": "Input.ChoiceSet",
+        "value": "0",
+        "id": "ignoreInput",
+        "isVisible": False,
+        "choices": [
+            {"title": "1 time", "value": "0"},
```

Comment:
@KKaempen there needs to be an "ignore immediately" option in addition to "ignore after x" occurrences.

Reply:
As it currently stands, the title of the card is "Ignore until this occurs again...". With the "immediately" option, that would read "Ignore until this occurs again immediately." Is that acceptable, or should I change the title/other options?

Reply:
@KKaempen Good question. I think having options that say "until this occurs x more times" is a bit confusing. What if we kept the title and just made the option "Ignore Immediately". I realize if you combine it with the title the sentence is kind of weird but I think it's clear to the user what the expected behavior is.

Reply:
Sounds good.

Reply:
What should the value be for that option?

Reply:
@KKaempen Can you actually change it to "ignore indefinitely" instead of "ignore immediately"? I realize that with all the options we are ignoring immediately which is confusing. Maybe a value of `-1` makes sense here if you can use numbers instead of strings.

Reply:
Going to make the value 0 (just the int) unless you disagree.

Reply:
Just saw this, yes I can do that.